### PR TITLE
[e2e] Added options to control e2e container execution 

### PR DESCRIPTION
**Pull request by adrianriobo**
This PR introduces two optional ENVS to control the execution of the e2e container:

* CLEANUP_HOME to control the deletion for crc home folder before running the test suite
* TESTING_MODE allowing to pick to modes: non-ux or ux

### CODE REVIEW

File: images/build-e2e/entrypoint.sh [hunk 1]
```diff
@@ -6,13 +6,19 @@ if [[ ${PLATFORM} == 'windows' ]]; then
...
+# Valid options are non-ux or ux
+TESTING_MODE="${TESTING_MODE:-"non-ux"}"
 # Review this when go 1.16 with embed support
 FEATURES_PATH=/opt/crc/features
 TESTDATA_PATH=/opt/crc/testdata
 UX_RESOURCES_PATH=/opt/crc/ux
+
 # Results
 RESULTS_PATH="${RESULTS_PATH:-/output}"
-CLEANUP_HOME="${CLEANUP_HOME:-"true"}"
```
Comment (cfergeau) [inline images/build-e2e/entrypoint.sh]: This move belongs to the previous commit

File: images/build-e2e/README.md [hunk 2]
```diff
@@ -6,41 +6,42 @@ The container connects through ssh to the target host and copy the right binary
...
-* RESULTS_PATH          Optional. Path inside container to pick results and logs from e2e execution.  
+**PLATFORM**:*define target platform (windows, macos, linux).*
+**TARGET_HOST**:*dns or ip for the target host.*  
+**TARGET_HOST_USERNAME**:*username for target host.*  
+**TARGET_HOST_KEY_PATH**:*private key for user. (Mandatory if not TARGET_HOST_PASSWORD).*  
+**TARGET_HOST_PASSWORD**:*password for user. (Mandatory if not TARGET_HOST_KEY_PATH).*  
+**PULL_SECRET_FILE_PATH** pull secret file path (local to container).*  
+**BUNDLE_VERSION**:*(Mandatory if not BUNDLE_LOCATION). Testing agaisnt crc released version bundle version for crc released version.*
+**BUNDLE_LOCATION**:*(Mandatory if not BUNDLE_VERSION). When testing crc with custom bundle set the bundle location on target server.*  
+**RESULTS_PATH**:*(Optional). Path inside container to pick results and logs from e2e execution.*
```
Comment (cfergeau) [inline images/build-e2e/README.md:17]: Fwiw, I'd do this reformatting, and the white space changes below in a separate commit so that it's easier to see the`CLEANUP_HOME`-related changes.

**Review by cfergeau — APPROVED** [review]
Except for some minor comments regarding the way the commits are split, this looks good to me.

File: images/build-e2e/entrypoint.sh (hunk 1 above)
Comment (adrianriobo) [reply images/build-e2e/entrypoint.sh]: Changed

File: images/build-e2e/README.md (hunk 2 above)
Comment (adrianriobo) [reply images/build-e2e/README.md:17]: Changed and splitted

File: images/build-e2e/entrypoint.sh [hunk 3]
```diff
@@ -72,8 +77,6 @@ else
 fi
 
 # Copy crc-e2e binary and pull-secret
-# Review this when ux feature cleanup environment 
-rm "${FEATURES_PATH}/ux.feature"
```
Comment (cfergeau) [inline images/build-e2e/entrypoint.sh:76]: Not clear why this change is in the `CLEANUP_HOME` commit?

File: images/build-e2e/README.md [hunk 4]
```diff
@@ -16,6 +16,7 @@ The container connects through ssh to the target host and copy the right binary
 **BUNDLE_LOCATION**:*(Mandatory if not BUNDLE_VERSION). When testing crc with custom bundle set the bundle location on target server.*  
 **RESULTS_PATH**:*(Optional). Path inside container to pick results and logs from e2e execution.*
 **CLEANUP_HOME**:*(Optional). Cleanup crc home folder or keep as it is to run test.*
+**TESTING_MODE**:*(Optinal). Define e2e testing mode (valid values are ux, non-ux. Default is non-ux)*
```
Comment (cfergeau) [inline images/build-e2e/README.md:19]: Typo, missing 'o' `Optional`

**Review by cfergeau — APPROVED** [review]

File: images/build-e2e/entrypoint.sh (hunk 3 above)
Comment (praveenkumar) [reply images/build-e2e/entrypoint.sh:76]: @adrianriobo this need to be resolved.

Comment (adrianriobo) [reply images/build-e2e/entrypoint.sh:76]: yeah, I am on it :+1:

Comment (adrianriobo) [reply images/build-e2e/entrypoint.sh:76]: moved to the testing_mode commit as this was related with the ux mode
